main/src/library/Reader.flix

```diff
@@ -27,5 +38,19 @@ trait Reader[t] {
     ///
     /// Returns `Err(e)` if some underlying I/O error occurs.
     ///
-    pub def read(b: Array[Int8, r], r: t): Result[IoError, Int32] \ {r, IO}
+    pub def read(b: Array[Int8, r], r: t): Result[IoError, Int32] \ {r, Reader.Eff[t]}
+
+}
+
+instance Reader[InputStream] {
+
+    type Eff = IO
+
+    pub def read(b: Array[Int8, r], r: InputStream): Result[IoError, Int32] \ {r, IO} =
```

Comment:
@mlutze Is this syntax considered deprecated or?

Reply:
I don't know that we made an official decision, but I prefer the effects written as:

```
r + IO
```

instead of

```
{r, IO}
```

The reason is that `r` is actually a set, so theoretically we should be `unioning` it with another set. Then `IO` is just syntactic sugar for `{IO}`

Reply:
Lets do r + IO (we can do it in the next PR)